modules/dataproc/main.tf

```diff
@@ -185,10 +185,10 @@ resource "google_dataproc_cluster" "cluster" {
         for_each = var.dataproc_config.cluster_config.dataproc_metric_config == null ? [] : [""]
         content {
           dynamic "metrics" {
-            for_each = var.dataproc_config.cluster_config.dataproc_metric_config.metrics == null ? [] : [""]
+            for_each = var.dataproc_config.cluster_config.dataproc_metric_config.metrics == null ? [] : var.dataproc_config.cluster_config.dataproc_metric_config.metrics
```

Comment:
nit: this is a coalesce

Reply:
You are right, i will change it with
`for_each = coalesce(var.dataproc_config.cluster_config.dataproc_metric_config.metrics, [])`